src/usdb_syncer/notes_parser.py

```diff
@@ -438,20 +453,104 @@ def minimum_song_length(self) -> str:
 
         return f"{minutes:02.0f}:{seconds:02.0f}"
 
-    def set_first_timestamp_to_zero(self) -> None:
+    def fix_first_timestamp(self) -> None:
         """Shifts all notes such that the first note starts at beat zero and adjusts
         GAP accordingly
         """
         if (offset := self.notes.start()) == 0:
+            # round GAP to nearest 10 ms
+            self.headers.gap = round(self.headers.gap / 10) * 10
             return
         for line in self.notes.all_lines():
             line.shift(-offset)
-        self.headers.gap = self.headers.gap + beats_to_ms(offset, self.headers.bpm)
+        self.headers.gap = (
+            round((self.headers.gap + beats_to_ms(offset, self.headers.bpm)) / 10) * 10
+        )
+
+    def fix_low_bpm(self) -> None:
+        """(repeatedly) multiplies BPM value and all note timings by two
+        until the BPM is in a range (200, 400)"""
+        bpm_threshold = 200.0
+
+        if (original_bpm := self.headers.bpm) >= bpm_threshold:
+            return
+
+        while self.headers.bpm < bpm_threshold:
+            self.headers.bpm *= 2
+
+        timing_factor = int(self.headers.bpm / original_bpm)
+
+        # modify medley tags
+        if self.headers.medleystartbeat:
+            self.headers.medleystartbeat *= timing_factor
+
+        if self.headers.medleyendbeat:
+            self.headers.medleyendbeat *= timing_factor
+
+        # modify all note timings
+        for line in self.notes.all_lines():
+            line.multiply(timing_factor)
+
+    def fix_line_breaks(self) -> None:
+        for i, line in enumerate(self.notes.player_1):
+            if line.line_break:
+                # remove end
+                line.line_break.end = None
+                # set line break timing similar to USDX implementation
+                next_line_start = self.notes.player_1[i + 1].start()
+                gap = next_line_start - line.end()
+
+                if gap < 2:
+                    line.line_break.start = next_line_start
+                elif gap == 2:
+                    line.line_break.start = line.end() + 1
+                else:
+                    line.line_break.start = line.end() + 2
+
+        if self.notes.player_2:
+            for i, line in enumerate(self.notes.player_1):
+                if line.line_break:
+                    # remove end
+                    line.line_break.end = None
+                    # set line break timing similar to USDX implementation
+                    next_line_start = self.notes.player_1[i + 1].start()
+                    gap = next_line_start - line.end()
+
+                    if gap < 2:
+                        line.line_break.start = next_line_start
+                    elif gap == 2:
+                        line.line_break.start = line.end() + 1
+                    else:
+                        line.line_break.start = line.end() + 2
```

Comment:
You can extract the for loop into a function and call it for both player_1 and player_2 to avoid the duplication.

If you want to use the `all_lines()` iterator, one option would be to store a refefernce to the last line:
```py
last_line = None
for line in self.notes.all_lines():
    if not (last_line and last_line.line_break):
        continue
    # fix break of last_line
    ...
    last_line = line
```